packages/alfa-rules/src/sia-er62/rule.ts

```diff
@@ -442,17 +421,26 @@ namespace Distinguishable {
     return (link) =>
       getForeground(container, device, context).some((containerColors) =>
         getForeground(link, device, context).some((linkColors) => {
-          const contrastValues = [
+          const contrastPairings: Contrast.Pairing[] = [
             ...Array.flatMap(containerColors, (containerColor) =>
               Array.map(linkColors, (linkColor) => {
-                return contrast(containerColor, linkColor);
+                return Contrast.Pairing.of(
+                  containerColor,
+                  linkColor,
+                  contrast(containerColor, linkColor)
+                );
               })
             ),
           ];
```

Comment:
That computation should be inside the nested `Cache.get` calls.
Currently, this always compute the pairing, then look in the cache and use the computation if the cache is empty. But the computation always happens…
The reason the second argument of `Cache.get` is a function is precisely to delay computation and don't do it if not needed.